### PR TITLE
Persistence types

### PR DESCRIPTION
**Pull request by marmistrz**
Depends on #1366.

### CODE REVIEW

File: golem/task/taskmanager.py [hunk 1]
```diff
@@ -458,7 +458,7 @@ def get_progresses(self):
         for t in list(self.tasks.values()):
             if t.get_progress() < 1.0:
                 ltss = LocalTaskStateSnapshot(t.header.task_id, t.get_total_tasks(),
-                                              t.get_active_tasks(), t.get_progress(), t.short_extra_data_repr(2200.0)) # FIXME in short_extra_data_repr should there be extra data
+                                              t.get_active_tasks(), t.get_progress(), t.short_extra_data_repr(2200.0))  # FIXME in short_extra_data_repr should there be extra data
```
Comment (inexxt) [inline golem/task/taskmanager.py]: Please fix length of those lines

**Review by inexxt — COMMENTED** [review]
Looks ok

File: golem/task/taskmanager.py (hunk 1 above)
Comment (marmistrz) [reply golem/task/taskmanager.py]: Fixed

**Comment by marmistrz** [comment]
Unfortunately, I don't know, how to fix that FIXME about extra_data. The class is defined in `taskbase.py` but it's really unobvious, how to convert float to `ExtraData`

**Comment by inexxt** [comment]
@badb What was the reason for introducing both `TaskManager.task_status` and `TaskState.status`?

**Review by inexxt — APPROVED** [review]

**Comment by inexxt** [comment]
EDIT: @marmistrz which `FIXME` are you talking about in https://github.com/golemfactory/golem/pull/1374#issuecomment-330790623?

**Comment by marmistrz** [comment]
@inexxt I don't really understand your comment. Can you be more precise?

**Comment by jiivan** [comment]
Closed in favor of #1581

**Pull request closed without merging**